test/e2e/ha/core_contrail_services_test.go

```diff
@@ -0,0 +1,445 @@
+package ha
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	contrail "github.com/Juniper/contrail-operator/pkg/apis/contrail/v1alpha1"
+	"github.com/Juniper/contrail-operator/pkg/client/kubeproxy"
+	"github.com/Juniper/contrail-operator/test/logger"
+	"github.com/Juniper/contrail-operator/test/wait"
+)
+
+var versionMap = map[string]string{
+	"cassandra":                     "3.11.4",
+	"zookeeper":                     "3.5.5",
+	"cemVersion":                    "2005.42",
+	"python":                        "3.8.2-alpine",
+	"redis":                         "4.0.2",
+	"busybox":                       "1.31",
+	"rabbitmq":                      "3.7",
+	"contrail-statusmonitor":        scmBranch + "." + scmRevision,
+	"contrail-operator-provisioner": scmBranch + "." + scmRevision,
+}
+
+func TestHACoreContrailServices(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+	log := logger.New(t, "contrail", test.Global.Client)
+
+	if err := test.AddToFrameworkScheme(contrail.SchemeBuilder.AddToScheme, &contrail.ManagerList{}); err != nil {
+		t.Fatalf("Failed to add framework scheme: %v", err)
+	}
+
+	if err := ctx.InitializeClusterResources(&test.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval}); err != nil {
+		t.Fatalf("Failed to initialize cluster resources: %v", err)
+	}
+	namespace, err := ctx.GetNamespace()
+	assert.NoError(t, err)
+	f := test.Global
+
+	proxy, err := kubeproxy.New(f.KubeConfig)
+	require.NoError(t, err)
+
+	t.Run("given contrail-operator is running", func(t *testing.T) {
```

Comment:
I would write Contrail Operator instead of hyphen-combined word

Reply:
done